Assets/Mirror/Runtime/QuaternionCompression.cs

```diff
@@ -0,0 +1,237 @@
+using UnityEngine;
+
+namespace Mirror
+{
+    /// <summary>
+    /// Functions to Compress Quaternions
+    /// </summary>
+    /// <remarks>
+    /// Uncompressed Quaternion = 32 * 4 = 128 bits => send 16 bytes
+    ///
+    /// <para>
+    ///     Quaternion is always normalized so we drop largest value and re-calculate it.
+    ///     We can encode which one is the largest using 2 bits
+    ///     <code>
+    ///     x^2 + y^2 + z^2 + w^2 = 1
+    ///     </code>
+    /// </para>
+    ///
+    /// <para>
+    ///     2nd largest value has max size of 1/sqrt(2)
+    ///     We can encode the smallest three components in [-1/sqrt(2),+1/sqrt(2)] instead of [-1,+1]
+    ///     <code>
+    ///     c^2 + c^2 + 0 + 0 = 1
+    ///     </code>
+    /// </para>
+    /// 
+    /// <para>
+    ///     Sign of largest value doesnt matter
+    ///     <code>
+    ///     Q * vec3 == (-Q) * vec3
+    ///     </code>
+    /// </para>
+    /// 
+    /// <list type="bullet">
+    /// <listheader><description>
+    ///     RotationPrecision <br/>
+    ///     <code>
+    ///     2/sqrt(2) / (2^bitCount - 1)
+    ///     </code>
+    /// </description></listheader>
+    /// 
+    /// <item><description>
+    ///     rotation precision +-0.00138 in range [-1,+1]
+    ///     <code>
+    ///     10 bits per value
+    ///     2 + 10 * 3 = 32 bits => send 4 bytes
+    ///     </code>
+    /// </description></item>
+    /// </list>
+    ///
+    /// <para>
+    /// Links for more info:
+    /// <br/><see href="https://youtu.be/Z9X4lysFr64">GDC Talk</see>
+    /// <br/><see href="https://gafferongames.com/post/snapshot_compression/">Post on Snapshot Compression</see>
+    /// </para>
+    /// </remarks>
+    public static class QuaternionCompression
+    {
+        const float MinValue = -1f / 1.414214f; // 1/ sqrt(2)
+        const float MaxValue = 1f / 1.414214f;
+        const float ValueRange = MaxValue - MinValue;
+
+        /// <summary>
+        /// Used to Compress Quaternion into 4 bytes
+        /// </summary>
+        public static uint Pack(Quaternion value)
+        {
+            value = value.normalized;
+
+            int largestIndex = FindLargestIndex(value);
```

Comment:
it seems like you missed the sign part.
"You might think you need to send a sign bit for w in case it is negative, but you don’t, because you can make w always positive by negating the entire quaternion if w is negative (in quaternion space (x,y,z,w) and (-x,-y,-z,-w) represent the same rotation.)"

Reply:
this is handled at the bottom of `GetSmallerDimensions`, I'm going to move it out so it is move obvious